packages/components/src/__future__/Tooltip/ToggleTip/ToggleTip.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { HTMLAttributes, ReactNode, useContext, useRef } from "react"
+import classnames from "classnames"
+import { AriaButtonOptions, useTooltipTrigger } from "react-aria"
+import { TooltipTriggerStateContext } from "react-aria-components"
+import styles from "./ToggleTip.module.scss"
+
+export type ToggleTipProps = {
+  children: ReactNode
+  className?: HTMLAttributes<HTMLButtonElement>["className"]
+} & AriaButtonOptions<"button">
+
+export const ToggleTip = ({
+  children,
+  className,
+  ...props
+}: ToggleTipProps): JSX.Element => {
+  const ref = useRef<HTMLButtonElement>(null)
+  const state = useContext(TooltipTriggerStateContext)!
+
+  // registers ref to triggerRef for positioning the tooltip
+  // it also registers event handler for escape key to close the tooltip
+  // other handlers like onKeyDown or onPointerDown we provide own
+  useTooltipTrigger(props, state, ref)
+
+  const toggle = (): void =>
+    state.isOpen ? state.close(true) : state.open(true)
+
+  return (
+    <button
+      ref={ref}
+      type="button"
+      className={classnames(styles.toggleTip, className)}
+      data-inline-hidden-content
+      onKeyDown={e => {
+        if (e.key === "Enter" || e.key === " ") toggle()
+      }}
+      onPointerDown={toggle}
+      onBlur={() => state.close(true)}
+      {...props}
+      // Cannot use react-aria's Button component as it prevents adding aria-hidden
+      aria-hidden="true"
```

Comment:
This seems like a critical issue that we should address, because it sounds like we are saying that `ToggleTip` needs to be a snowflake button and as such will not behave consistently with our main `Button` component (or inherit any updates).

I also think this might be unnatural in our consumer's workflow, we'd need to check with Jess, but I imagine designers will simply drag a Button and a Tooltip onto their canvas and say "this should toggle the Tooltip", without even thinking to search for a specific component -- given how many ToggleTips/Tooltips are present in the existing product, this could cause some friction around:

- Button features not being present in `ToggleTip` in which devs/design will pull us up on that
- Devs struggling to assemble the `Tooltip` and `Button` only to either ask or realise late in the process that there's a `ToggleTip` component, which then brings up the first point.

But given we cannot actually attach `Tooltip` to the official `Button`, both of these points potentially have a high chance of resulting in a bespoke component being made, slowing down consumers for a very common pattern in  our products.

So I think we need to discuss this one and consider what we might do to avoid these scenarios since it will undoubtedly affect adoption rates.